nosql/nosql.functions

```diff
@@ -0,0 +1,406 @@
+				comment_text: "brilliant speech!"
+			}
+		}
+	}
+)
+
+
+db.posts.insert({
+	created_by: ObjectId("57c279ceb5f524417913f9a9"),
+	posted_at: new Date(),
+	heading: "Canon 5D Mark IV DSLR Announced",
+	body: "After way too long of a wait, we finally have a sequel from the " + 
+		  "famed Canon 5D Mark III. While Canon has enjoyed plenty of" + 	
+		  "success with that camera, it certainly has felt a little old" + 
+		  "in recent months, especially in comparison to the newer mirrorless Sony" +
+		  "A7RII and Nikon D810 DSLR. Thankfully, we have quite a few new things" +
+		  "to get excited over in the Canon 5D Mark IV so let's take a look.",
+	photos: [
+		{
+			url: "http://somephoto.url.jpg",
+			caption: "Canon 5D Mark IV DSLR"
+		}
+	],
+	hash_tags: [
+		"tag text",
+	]
+})
+
+db.posts.insert({
+	created_by: ObjectId("57c279ceb5f524417913f9a9"),
+	posted_at: new Date(),
+	heading: "How to Get the Most Value Out of Your Coding Bootcamp",
+	body: "This depends on what your goal is for going to a coding bootcamp, but I'll assume for now that it's to get a job as a software engineer. One of the biggest things I've seen in outcomes at Fullstack is that attending the program acts as a multiplier of your skills/knowledge.",
+	photos: [
+		{
+			url: "http://somephoto.url.jpg",
+			caption: "Canon 5D Mark IV DSLR"
+		},
+		{
+			url: "http://somephoto.url2.jpg",
+			caption: "Canon 5D Mark IV DSLR"
+		}
+	]
+})
+
+
+
+db.posts.find({created_by : ObjectId("57c279ceb5f524417913f9a9")})
+
+db.users.find(
+	{
+		_id: ObjectId("57c279ceb5f524417913f9a9")
+	}
+)
+
+db.users.update(
+	{
+		_id: ObjectId("57c279b1b5f524417913f9a7")
+	},
+	{
+		$set: {
+			"profile.name": {
+				first_name: "Shirish",
+				second_name: "Shrikant",
+				last_name: "Parsekar"
+			}
+		}
+	}
+)
+
+db.users.update(
+	{
+		_id: ObjectId("57c279b1b5f524417913f9a7")
+	},
+	{
+		$set: {
+			"profile.address": {
+				state: "Goa",
+				country: "India"
+			}
+		}
+	}
+)
+
+db.users.update(
+	{
+		_id: ObjectId("57c279b1b5f524417913f9a7")
+	},
+	{
+		$set: {
+			"profile.date_of_birth": {
+				day: "14",
+				month: "09",
+				year: "1992"
+			}
+		}
+	}
+)
+db.users.update(
+	{
+		_id: ObjectId("57c279b1b5f524417913f9a7")
+	},
+	{
+		$set: {
+			"profile.about_me": 
+				"i am this,"+
+				"i am this,"+
+				"i am this"
+		}
+	}
+)
+
+
+db.posts.find(
+	{
+		_id: ObjectId("57c27b1d9069c2163b338378")
+	},
+	{
+		_id: 0,
+		comments: 1,
+
+	}
+).sort({"comments.posted_at" : -1}).pretty()
+
+db.posts.update(
+	{
+		_id: ObjectId("57c27b1d9069c2163b338378"),
+	},
+	{
+		$push : { 
+			comments: {
+				comment_id: ObjectId(),
+				posted_at: new Date(),
+				user_id: ObjectId("57bebcc39ff2d15a38d6a62c"),
+				display_name: "rahul007",
+				comment_text: "brilliant speech!"
+			}
+		}
+	}
+)
+
+
+--------------------------------------------------------------------------
+
+ Review queries
+
+--------------------------------------------------------------------------
+
+db.posts.find(
```

Comment:
When you are fetching a single post specially with _id, use findOne() over find()